src/components/data-grid/export/docDefinition.ts

```diff
@@ -0,0 +1,250 @@
+import { ColumnApi, GridApi, ValueFormatterParams } from '@ag-grid-community/core';
+import { Content, Margins, TDocumentDefinitions } from 'pdfmake/interfaces';
+import { PdfHeaderCell, PdfTableCell, PrintParams, Translations } from '../types';
+
+export const getDocDefinition = (
+  gridApi: GridApi,
+  columnApi: ColumnApi,
+  printParams: PrintParams,
+  translations: Translations,
+) => {
+  const {
+    PDF_HEADER_COLOR,
+    PDF_INNER_BORDER_COLOR,
+    PDF_OUTER_BORDER_COLOR,
+    PDF_ODD_BKG_COLOR,
+    PDF_EVEN_BKG_COLOR,
+    PDF_HEADER_HEIGHT = 25,
+    PDF_ROW_HEIGHT = 15,
+    PDF_PAGE_ORIENTATION,
+    PDF_WITH_HEADER_IMAGE,
+    PDF_WITH_FOOTER_PAGE_COUNT,
+    PDF_LOGO = '',
+  } = printParams;
+
+  return (function () {
+    const columnGroupsToExport = getColumnGroupsToExport();
+
+    const columnsToExport = getColumnsToExport();
+
+    const widths = getExportedColumnsWidths(columnsToExport);
+
+    const rowsToExport = getRowsToExport(columnsToExport);
+
+    const body = columnGroupsToExport
+      ? [columnGroupsToExport, columnsToExport, ...rowsToExport]
+      : [columnsToExport, ...rowsToExport];
+
+    const headerRows = columnGroupsToExport ? 2 : 1;
+
+    const header = PDF_WITH_HEADER_IMAGE
+      ? ({
+          image: 'ag-grid-logo',
+          width: 150,
+          alignment: 'center',
+          margin: [0, 10, 0, 10],
+        } as Content)
+      : undefined;
+
+    const footer = PDF_WITH_FOOTER_PAGE_COUNT
+      ? function (currentPage: number, pageCount: number) {
+          const footer: Content = {
+            text: translations.paginationOutOfLong(currentPage, pageCount),
+            margin: [20, 20, 20, 20],
+          };
+          return footer;
+        }
+      : undefined;
+
+    const pageMargins: Margins = [10, PDF_WITH_HEADER_IMAGE ? 70 : 20, 10, PDF_WITH_FOOTER_PAGE_COUNT ? 40 : 10];
+
+    const heights = (rowIndex: number) => (rowIndex < headerRows ? PDF_HEADER_HEIGHT : PDF_ROW_HEIGHT);
+
+    const fillColor = (rowIndex: number, node: any) => {
+      if (rowIndex < node.table.headerRows) {
+        return PDF_HEADER_COLOR;
+      }
+      return rowIndex % 2 === 0 ? PDF_ODD_BKG_COLOR : PDF_EVEN_BKG_COLOR;
+    };
+
+    const hLineWidth = (i: number, node: any) => (i === 0 || i === node.table.body.length ? 1 : 1);
```

Comment:
opinion-based question: would it be better to prefix variables with get- so that it's clearer that we're dealing with functions?